search/django/weedid/tasks.py

```diff
@@ -0,0 +1,58 @@
+from __future__ import absolute_import, unicode_literals
+from celery import shared_task
+from weedcoco.repo.deposit import deposit
+from weedcoco.index.indexing import ElasticSearchIndex
+from weedcoco.index.thumbnailing import thumbnailing
+from weedid.models import Dataset
+from core.settings import THUMBNAILS_DIR, REPOSITORY_DIR, DOWNLOAD_DIR
+from pathlib import Path
+
+
+@shared_task
+def upload_task(weedcoco_path, image_dir, upload_id):
+    upload_entity = Dataset.objects.get(upload_id=upload_id)
+    upload_entity.upload_status = "P"
+    upload_entity.upload_status_details = ""
+    upload_entity.save()
+    try:
```

Comment:
We need to precede deposit with:
1. Convert to weedcoco (when input is not weedcoco)
2. Validate weedcoco